src/js/view/controls/components/menu/tizen-menu.js

```diff
@@ -0,0 +1,231 @@
+            case 38: // Up
+                onNavigateMenu(true);
+                break;
+            case 40: // Down
+                onNavigateMenu(false);
+                break;
+            case 37: // Left
+                this.close();
+                if (this.controlbar) {
+                    this.controlbar.elements.settingsButton.element().focus();
+                }
+                break;
+            case 415: // Play
+            case 10252: // Play/Pause
+                this.close(evt, true);
+                this.api.play({ reason: 'settingsInteraction' });
+                break;
+            case 10009: // Back
+                this.close(evt);
+                break;
+            default:
+                break;
+        }
+    }
+
+    setVisibility(evt) {
+        const el = this.el;
+        
+        if (evt.visible) {
+            this.api.pause({ reason: 'settingsInteraction' });
+            document.addEventListener('keydown', this.onInteraction);
+            el.classList.remove('jw-settings-transition-close');
+            el.classList.add('jw-settings-open');
+            el.classList.add('jw-settings-transition-open');
+            el.setAttribute('aria-expanded', 'true');
+            this.visible = true;
+            const menuNames = Object.keys(this.children);
+            if (this.children && menuNames.length) {
+                // Focus first item of first menu
+                this.children[menuNames[0]].items[0].el.focus({ preventScroll: true });
+            }
+        } else {
+            document.removeEventListener('keydown', this.onInteraction);
+            el.classList.remove('jw-settings-transition-open');
+            el.classList.add('jw-settings-transition-close');
+            el.setAttribute('aria-expanded', 'false');
+            this.visible = false;
+        }
+
+        if (this.controlbar) {
+            // Hide controlbar when showing menu
+            this.controlbar.toggleVisibility(!evt.visible);
+        }
+    }
+
+    open(evt) {
+        if (this.visible) {
+            return;
+        }
+        this.trigger('visibility', { visible: true, evt });
+    }
+
+    close(evt) {
+        if (!this.visible) {
+            return;
+        }
+        this.trigger('visibility', { visible: false, evt });
+    }
+
+    toggle(evt) {
+        this.trigger('visibility', { visible: !this.visible, evt });
+    }
+
+    onCaptionsList(model, captionsList) {
+        const menuItemOptions = { defaultText: this.localization.off };
+        const initialIndex = model.get('captionsIndex');
+        
+        this.setupMenu(
+            'captions',
+            this.localization.cc,
+            captionsList, 
+            (index) => this.api.setCurrentCaptions(index), 
+            initialIndex, 
+            menuItemOptions
+        );
+    }
+
+    onCaptionsIndex(model, index) {
+        const captionsMenu = this.children.captions;
+        if (captionsMenu) {
```

Comment:
No issue, just wondering if we wanted to short circuit this and return if there is no captionsMenu like we do below with the audioTracksMenu?